specification/cognitiveservices/data-plane/Language/stable/2022-05-01/conversationalanalysisauthoring.json

```diff
@@ -0,0 +1,3060 @@
+          "$ref": "#/definitions/ConversationalAnalysisAuthoringCompositionSetting"
+        },
+        "list": {
+          "$ref": "#/definitions/ConversationalAnalysisAuthoringExportedEntityList"
+        },
+        "prebuilts": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConversationalAnalysisAuthoringExportedPrebuiltEntity"
+          }
+        }
+      },
+      "x-ms-client-name": "ConversationExportedEntity"
+    },
+    "ConversationalAnalysisAuthoringConversationExportedIntent": {
+      "required": [
+        "category"
+      ],
+      "type": "object",
+      "properties": {
+        "category": {
+          "type": "string"
+        }
+      },
+      "x-ms-client-name": "ConversationExportedIntent"
+    },
+    "ConversationalAnalysisAuthoringConversationExportedProjectAssets": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ConversationalAnalysisAuthoringExportedProjectAssets"
+        }
+      ],
+      "properties": {
+        "intents": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConversationalAnalysisAuthoringConversationExportedIntent"
+          }
+        },
+        "entities": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConversationalAnalysisAuthoringConversationExportedEntity"
+          }
+        },
+        "utterances": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConversationalAnalysisAuthoringConversationExportedUtterance"
+          }
+        }
+      },
+      "x-ms-discriminator-value": "conversation",
+      "x-ms-client-name": "ConversationExportedProjectAssets"
+    },
+    "ConversationalAnalysisAuthoringConversationExportedUtterance": {
+      "required": [
+        "text",
+        "intent"
+      ],
+      "type": "object",
+      "properties": {
+        "entities": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ConversationalAnalysisAuthoringExportedUtteranceEntityLabel"
+          }
+        },
+        "text": {
+          "type": "string"
+        },
+        "language": {
+          "type": "string"
+        },
+        "intent": {
+          "type": "string"
+        },
+        "dataset": {
+          "description": "The dataset for this utterance. Allowed values are 'Train' and 'Test'.",
+          "type": "string"
+        }
+      },
+      "x-ms-client-name": "ConversationExportedUtterance"
+    },
+    "ConversationalAnalysisAuthoringCreateDeploymentOptions": {
+      "required": [
+        "trainedModelLabel"
+      ],
+      "type": "object",
+      "properties": {
+        "trainedModelLabel": {
+          "description": "Represents the trained model label.",
+          "type": "string"
+        }
+      },
+      "x-ms-client-name": "CreateDeploymentOptions"
+    },
+    "ConversationalAnalysisAuthoringCreateProjectOptions": {
```

Comment:
This is really the same as `ConversationalAnalysisAuthoringProjectMetadata` and should be "folded" into a single model with `readOnly` properties. Unless I'm missing something?

Reply:
Never mind, I was missing something. There is a difference in requiredness between the create and read models - which, in turn, is due to the use of PATCH for upsert.

Reply:
Actually, you're right. They are the same, except for the "DateTime" properties (E.g. `lastModifiedDateTime`). In the PATCH body, the properties are marked "required" assuming that it's a Create request (i.e. we're patching an empty object).

Should we make these properties optional, given that it's a merge-patch+json request, or should we mark them as required assuming that the resource didn't exist before this request?

Reply:
And just to make sure that I understand correctly. We can mark properties as `readonly` to indicate that it's not a user input, and it's a server generated value. This applies to generated `DateTime` and maybe the `modelId` (system generated).

Or asking the question in a different way. Should we use `readonly` for all the responses that we return if it's not updatable? E.g. The evaluation results are responses only, should we make all its properties `readonly`?